src/DefinitionResolver.php

```diff
@@ -667,12 +666,26 @@ public function resolveExpressionNodeToType($expr)
                 } else {
                     $classFqn = substr((string)$t->getFqsen(), 1);
                 }
-                $fqn = $classFqn . '->' . $expr->memberName->getText($expr->getFileContents());
+                $add = '->' . $expr->memberName->getText($expr->getFileContents());
                 if ($expr->parent instanceof Node\Expression\CallExpression) {
-                    $fqn .= '()';
+                    $add .= '()';
                 }
-                $def = $this->index->getDefinition($fqn);
-                if ($def !== null) {
+                $lookupDefinition = function (string $base, string $add) use (&$lookupDefinition) {
```

Comment:
Not really - using a generator and looking for the exact method definition guarantees that the execution will break as soon as a match is found. 
Using ```CompletionProvider::expandParentFqns()``` means looking up the whole class hierarchy, storing all definitions in an array, then iterating over it to find matches for the method signature (more definition lookups).

Of course one might consider my approach as premature optimization?